Lib/curses/textpad.py

```diff
@@ -81,7 +81,7 @@ def _insert_printable_char(self, ch):
                 self.win.addch(ch)
             except curses.error:
                 pass
-            if not self.insert_mode or not curses.ascii.isprint(oldch):
+            if not self.insert_mode and not curses.ascii.isprint(oldch):
```

Comment:
The test run fails on this line due to ``oldch`` not being set. That's because ``oldch`` is only set on line 76 when ``self.insert_mode`` is true.

Reply:
I agree with you. since `oldch` it's not really necessary in that case I can simply define it in line 73 to None, something like:
```python
# Line 73
oldch = backyx = None
```
Just to avoid those test failures, results it's almost the same. If you agree I can change my MR with it. What do you think ?

Reply:
Good question, I'll have to do some research on this as I'm not familiar with this part of the code base.